test/java/seedu/addressbook/commands/ViewCommandTest.java

```diff
@@ -0,0 +1,159 @@
+                                            new Email("some@hey.go", true),
+                                            new Address("nus", false),
+                                            new UniqueTagList(Collections.emptySet()));
+        listWithAll = new ArrayList<ReadOnlyPerson>(listWithAll);
+        listWithAll.add(someone);
+
+        // empty addressbook
+        assertViewErrorPersonNotInAddressBook(emptyAddressBook, listWithAll, 1);
+
+        // non-empty addressbook
+        assertViewErrorPersonNotInAddressBook(addressBook, listWithAll, listWithAll.size());
+    }
+
+    @Test
+    public void viewCommand_validIndex_returnsPersonDetails() {
+        // person with no private information
+        assertViewSuccess(addressBook, listWithAll, 1);
+
+        // person with some private information
+        assertViewSuccess(addressBook, listWithAll, 2);
+
+        // person with all private information
+        assertViewSuccess(addressBook, listWithAll, 4);
+
+        // addressbook has more people than displayList
+        // This can happen when view from filtered list caused by some commands(eg. FindCommand)
+        assertViewSuccess(addressBook, listWithSome, 1);
+    }
+
+    private Command generateViewCommand(AddressBook addressBook, List<ReadOnlyPerson> displayList, int index) {
+        ViewCommand command = new ViewCommand(index);
+        command.setData(addressBook, displayList);
+
+        return command;
+    }
+
+    private Command generateViewAllCommand(AddressBook addressBook, List<ReadOnlyPerson> displayList, int index) {
+        Command command = new ViewAllCommand(index);
+        command.setData(addressBook, displayList);
+
+        return command;
+    }
+
+    /**
+     * Asserts that both a ViewCommand and a ViewAllCommand can retrieve from 
+     * the {@code addressBook} details of the person at the given {@code index} in the given {@code list}.
+     * 
+     * @param index one-indexed position of the target person in the list
+     */
+    private void assertViewSuccess(AddressBook addressBook, List<ReadOnlyPerson> list, int index) {
+        ReadOnlyPerson personToBeViewed = list.get(index - 1); // -1 is because index is one-indexed
+
+        String expectedMessage = String.format(ViewCommand.MESSAGE_VIEW_PERSON_DETAILS,
+                                               personToBeViewed.getAsTextHidePrivate());
+        assertViewCommandBehaviour(addressBook, list, index, expectedMessage);
+
+        expectedMessage = String.format(ViewCommand.MESSAGE_VIEW_PERSON_DETAILS,
+                                        personToBeViewed.getAsTextShowAll());
+        assertViewAllCommandBehaviour(addressBook, list, index, expectedMessage);
+    }
+
+    /**
+     * Asserts that the details of person at specific index cannot be retrieved due to
+     * invalid index.
+     */
+    private void assertViewErrorInvalidIndex(AddressBook addressBook, List<ReadOnlyPerson> list, int index) {
```

Comment:
Does this really need its own method?

Yes, I know the alternative is repeating `Messages.MESSAGE_INVALID_PERSON_DISPLAYED_INDEX` in `viewCommand_invalidIndex_returnsInvalidIndexMessage()`, but then again with this method you still need to repeat `assertViewErrorInvalidIndex` anyway, which probably means this method doesn't add much value.

If the reason is because `Messages.MESSAGE_INVALID_PERSON_DISPLAYED_INDEX` is too long, then you can assign in to a variable with a shorter name.